runtime/libia2/include/ia2_internal.h

```diff
@@ -40,29 +47,46 @@ struct dl_phdr_info;
+#define REPEATA4(fn, basefn, ...) REPEATA3(fn, basefn, ##__VA_ARGS__) fn(4, ##__VA_ARGS__)
+#define REPEATA5(fn, basefn, ...) REPEATA4(fn, basefn, ##__VA_ARGS__) fn(5, ##__VA_ARGS__)
+#define REPEATA6(fn, basefn, ...) REPEATA5(fn, basefn, ##__VA_ARGS__) fn(6, ##__VA_ARGS__)
+#define REPEATA7(fn, basefn, ...) REPEATA6(fn, basefn, ##__VA_ARGS__) fn(7, ##__VA_ARGS__)
+#define REPEATA8(fn, basefn, ...) REPEATA7(fn, basefn, ##__VA_ARGS__) fn(8, ##__VA_ARGS__)
+#define REPEATA9(fn, basefn, ...) REPEATA8(fn, basefn, ##__VA_ARGS__) fn(9, ##__VA_ARGS__)
+#define REPEATA10(fn, basefn, ...) REPEATA9(fn, basefn, ##__VA_ARGS__) fn(10, ##__VA_ARGS__)
+#define REPEATA11(fn, basefn, ...) REPEATA10(fn, basefn, ##__VA_ARGS__) fn(11, ##__VA_ARGS__)
+#define REPEATA12(fn, basefn, ...) REPEATA11(fn, basefn, ##__VA_ARGS__) fn(12, ##__VA_ARGS__)
+#define REPEATA13(fn, basefn, ...) REPEATA12(fn, basefn, ##__VA_ARGS__) fn(13, ##__VA_ARGS__)
+#define REPEATA14(fn, basefn, ...) REPEATA13(fn, basefn, ##__VA_ARGS__) fn(14, ##__VA_ARGS__)
+#define REPEATA15(fn, basefn, ...) REPEATA14(fn, basefn, ##__VA_ARGS__) fn(15, ##__VA_ARGS__)
+#define REPEATB0(fn, basefn, ...) basefn(0, ##__VA_ARGS__)
+#define REPEATB1(fn, basefn, ...) fn(1, ##__VA_ARGS__) REPEATB0(fn, basefn, ##__VA_ARGS__)
+#define REPEATB2(fn, basefn, ...) fn(2, ##__VA_ARGS__) REPEATB1(fn, basefn, ##__VA_ARGS__)
+#define REPEATB3(fn, basefn, ...) fn(3, ##__VA_ARGS__) REPEATB2(fn, basefn, ##__VA_ARGS__)
+#define REPEATB4(fn, basefn, ...) fn(4, ##__VA_ARGS__) REPEATB3(fn, basefn, ##__VA_ARGS__)
+#define REPEATB5(fn, basefn, ...) fn(5, ##__VA_ARGS__) REPEATB4(fn, basefn, ##__VA_ARGS__)
+#define REPEATB6(fn, basefn, ...) fn(6, ##__VA_ARGS__) REPEATB5(fn, basefn, ##__VA_ARGS__)
+#define REPEATB7(fn, basefn, ...) fn(7, ##__VA_ARGS__) REPEATB6(fn, basefn, ##__VA_ARGS__)
+#define REPEATB8(fn, basefn, ...) fn(8, ##__VA_ARGS__) REPEATB7(fn, basefn, ##__VA_ARGS__)
+#define REPEATB9(fn, basefn, ...) fn(9, ##__VA_ARGS__) REPEATB8(fn, basefn, ##__VA_ARGS__)
+#define REPEATB10(fn, basefn, ...) fn(10, ##__VA_ARGS__) REPEATB9(fn, basefn, ##__VA_ARGS__)
+#define REPEATB11(fn, basefn, ...) fn(11, ##__VA_ARGS__) REPEATB10(fn, basefn, ##__VA_ARGS__)
+#define REPEATB12(fn, basefn, ...) fn(12, ##__VA_ARGS__) REPEATB11(fn, basefn, ##__VA_ARGS__)
+#define REPEATB13(fn, basefn, ...) fn(13, ##__VA_ARGS__) REPEATB12(fn, basefn, ##__VA_ARGS__)
+#define REPEATB14(fn, basefn, ...) fn(14, ##__VA_ARGS__) REPEATB13(fn, basefn, ##__VA_ARGS__)
+#define REPEATB15(fn, basefn, ...) fn(15, ##__VA_ARGS__) REPEATB14(fn, basefn, ##__VA_ARGS__)
 /* clang-format on */
 
 /* Macro to repeatedly apply a function or function-like macro `fn` a given
 number of times, passing the index to each invocation. The passed index `n` is
 first, followed by n-1 and so on. For the base case of 0, `basefn` is applied
 instead of `fn`. */
-#define REPEATB(n, fn, basefn) REPEATB##n(fn, basefn)
+#define REPEATA(n, fn, basefn, ...) REPEATA##n(fn, basefn, ##__VA_ARGS__)
```

Comment:
Can you document this? As a historical note, the `REPEATB` name came from "**REPEAT** this macro application _n_ times, with a different **B**ase case". What's the intuition for the `REPEATA` name and how is it used? Is it just expanding in the opposite order to give an ascending sequence?